addons/sale_project/tests/test_sale_project_dashboard.py

```diff
@@ -0,0 +1,124 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.sale_project.tests.test_project_profitability import TestProjectProfitabilityCommon as Common
+
+
+class TestProjectDashboardCommon(Common):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.dashboard_project = cls.env['project.project'].with_context({'mail_create_nolog': True}).create({
+            'name': 'Project',
+            'partner_id': cls.partner.id,
+            'analytic_account_id': cls.analytic_account.id,
+            'allow_billable': True,
+        })
+        cls.dashboard_sale_order = cls.env['sale.order'].with_context(tracking_disable=True).create({
+            'partner_id': cls.partner.id,
+            'partner_invoice_id': cls.partner.id,
+            'partner_shipping_id': cls.partner.id,
+        })
+        cls.dashboard_sale_order.action_confirm()
+        cls.dashboardSaleOrderLine = cls.env['sale.order.line'].with_context(tracking_disable=True, default_order_id=cls.dashboard_sale_order.id)
+        cls.dashboard_product_delivery_service = cls.env['product.product'].create({
```

Comment:
I guess you could pack the 3 following writes in a single write and use tuple unpacking, but yeah. It'd be ugly and I guess that's why you didn't do it

Reply:
yup I'd rather not change it, i think it's more readable this way. But to be honest, i did not even think about using tuple unpacking.

Reply:
Actually, it is better to create in batch for performance wise, I understand it could be ugly to unpack. 😁